### ISSUE TITLE
Add new menu layouts

### ISSUE THREAD

**Issue by orhun** [orhun/kmon#29]
**Is your feature request related to a problem? Please describe.**
It would be nice to have different menu layouts instead of having only one which is fixed. We already have the functionality for changing the block sizes and moving them around. This new menu(s) could be possibly change the overall layout for different use cases such as:

- Kernel monitoring (dmesg) mode where the "Kernel Activities" block is maximized.
- Inspection mode where only the module info is shown.

And a couple of other scenarios...

**Describe the solution you'd like**
Define a list of menu layouts and add a key binding for changing the current layout.

**Describe alternatives you've considered**
There could be a command-line argument for launching `kmon` with the given menu type. e.g. `kmon --menu activities`

**Additional context**
[bpytop](https://github.com/aristocratos/bpytop) has something similar where you press `m` and the layout changes.

**Comment by Muhammad-Owais-Warsi** [comment]
I would love to contribute, but need a lot of help, as I have no idea on how to implement this. So any similar resources and docs would be of great help.

**Comment by orhun** [comment]
@Muhammad-Owais-Warsi great to hear, I'm happy to help :)

You can start of by reading the Ratatui documentation/templates to get an idea how the project is structured: <https://ratatui.rs/> ([templates](https://github.com/ratatui/templates))

Then taking a look at the `app.rs` file would be a good idea. That's simply where the TUI state/drawing happens.

And then you can give a shot implementing it, ask any questions!

**Comment by orhun** [comment]
It depends on which constrains you are changing. Those seems to be the Table constraints, I think you need to look into `lib.rs` for the layout constraints.

**Comment by Muhammad-Owais-Warsi** [comment]
In the `lib.rs` file on changing the constraints in [lib.rs](https://github.com/orhun/kmon/blob/master/src/lib.rs), for example on line number 42, it is written as `Constraint::Percentage(100 - app.block_size.activities)` which gives me the following output:


![Screenshot from 2024-10-16 02-35-26](https://github.com/user-attachments/assets/12c43d0d-c309-4a6c-bf2f-8055feb7ad1b)

Changing the same line to `Constraint::Percentage(50)` gives me the following output:

![Screenshot from 2024-10-16 02-35-56](https://github.com/user-attachments/assets/001bc088-a0a3-46f6-ba15-0215a28ebfac)

Is this what we're trying to solve as per the issue? If I’m not mistaken, we also need to implement a key binding so that we can switch between different layouts. Let me know if I misunderstood something.

Thanks :)

**Comment by orhun** [comment]
Yes, we want to change the constraints and also the order of the layouts. If you play around with the tool a bit, you might realize that sometimes the text does not fit the exact area - we want to provide multiple layout options for improving the user experience in those cases.

Take a look at [bpytop](https://github.com/aristocratos/bpytop?tab=readme-ov-file#screenshots)'s different UI modes (mini, detailed, etc) for inspiration :)

**Comment by Muhammad-Owais-Warsi** [comment]
> Yes, we want to change the constraints and also the order of the layouts. If you play around with the tool a bit, you might realize that sometimes the text does not fit the exact area - we want to provide multiple layout options for improving the user experience in those cases.
> 
> Take a look at [bpytop](https://github.com/aristocratos/bpytop?tab=readme-ov-file#screenshots)'s different UI modes (mini, detailed, etc) for inspiration :)

Thanks, I'll look into it.
